include/rrc/core/AbstractPropertyListener.h

```diff
@@ -5,14 +5,33 @@
 #pragma once
 
 #include "Property.h"
+#include <map>
 
 namespace rrc {
+    /**
+     * @brief Abstract class for getting needed up-to-date collection of settings from the settings billboard; also can get needed property by it's name
+     */
     class AbstractPropertyListener {
     public:
 
-        virtual void setProperty(Property property) = 0;
+        /**
+         * @brief Returns property with the specified name
+         * @param key Name of the needed property
+         * @return Property
+         */
+        virtual Property getProperty(std::string key) = 0;
 
-        virtual Property getProperty() = 0;
+        /**
+         * @brief Sets the collection of settings to this listener
+         * @param dictionary CopyOnWrite map with the needed settings
+         */
+        virtual void setDictionary(CopyOnWrite<std::map<std::string, Property>> dictionary) = 0;
```

Comment:
You shoud be create PropertyDictionary wrapper for CopyOnWrite<std::map<std::string, Property>>